docs/development/go/developing-udp-and-tcp-clients-and-servers-in-go/index.md

```diff
@@ -0,0 +1,579 @@
+listen to. Valid values for the first parameter are `tcp`, `tcp4` (IPv4-only), `tcp6` (IPv6-only),
+`udp`, `udp4` (IPv4-only), `udp6` (IPv6-only), `ip`, `ip4` (IPv4-only), `ip6` (IPv6-only),
+`Unix` (Unix sockets), `Unixgram` and `Unixpacket`.
+
+Apart from the `net.Listen()` function, there exist `net.ListenUDP()` and `net.ListenTCP()`, which
+are for creating UDP and TCP servers, respectively. These two functions should be paired with
+`net.ResolveUDPAddr()` and `net.ResolveTCPAddr()`, respectively.
+
+Similarly, the parameters of the `net.Dial()`, `net.DialTCP()` and `net.DialUDP()` specify the
+kind of client you are going to create. The `net.Dial()` function is the most generic one because
+it can create all kinds of network clients whereas `net.DialTCP()` and `net.DialUDP()` can
+create TCP and UDP clients, respectively.
+
+The TCP part of this guide will use the generic functions whereas the UDP part will use
+`net.ResolveUDPAddr()`, `net.ListenUDP()` and `net.DialUDP()`.
+
+{{< note >}}
+All versions of `net.Dial()` and `net.Listen()` return data types that implement the
+`io.Reader` and `io.Writer` interfaces. This means that you can use regular File I/O
+functions to send and receive data from a TCP/IP connection if you want.
+{{< /note >}}
+
+{{< note >}}
+If you ever want to test a TCP/IP application, either a server or a client, you might
+find the `nc(1)` command line utility very handy.
+{{< /note >}}
+
+## A TCP Client
+
+In this section of the guide, we are going to see how to develop a generic TCP client in Go.
+The utility will allow you to interact with any TCP server.
+
+### Looking at the Go code of the TCP client
+
+The Go code of the TCP client is the following:
+
+{{< file "./tcpC.go" go >}}
+package main
+
+import (
+        "bufio"
+        "fmt"
+        "net"
+        "os"
+        "strings"
+)
+
+func main() {
+        arguments := os.Args
+        if len(arguments) == 1 {
+                fmt.Println("Please provide host:port.")
+                return
+        }
+
+        CONNECT := arguments[1]
+        c, err := net.Dial("tcp", CONNECT)
+        if err != nil {
+                fmt.Println(err)
+                return
+        }
+
+        for {
+                reader := bufio.NewReader(os.Stdin)
+                fmt.Print(">> ")
+                text, _ := reader.ReadString('\n')
+                fmt.Fprintf(c, text+"\n")
+
+                message, _ := bufio.NewReader(c).ReadString('\n')
+                fmt.Print("->: " + message)
+                if strings.TrimSpace(string(text)) == "STOP" {
+                        fmt.Println("TCP client exiting...")
+                        return
+                }
+        }
+}
+{{< /file >}}
+
+ - The first part of the `main()` function gathers command line arguments and makes sure that `host:port` was sent. This argument is then put into the `CONNECT` variable to be used in the `net.Dial()` call. The implementation of the TCP client begins
```

Comment:
I changed this - it said that this had nothing to do with TCP, which isn't true. It's getting the host:port here.